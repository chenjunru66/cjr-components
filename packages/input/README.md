<!--
 * @Author: chenjunru
 * @Date: 2022-12-25 14:45:35
 * @LastEditTime: 2022-12-25 17:46:43
 * @LastEditors: chenjunru
 * @Description: Do not Edit
 * @FilePath: \my-components\packages\input\README.md
-->
# CjrInput