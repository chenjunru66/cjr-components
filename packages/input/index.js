/*
 * @Author: chenjunru
 * @Date: 2022-12-25 14:45:35
 * @LastEditTime: 2022-12-25 17:45:49
 * @LastEditors: chenjunru
 * @Description: Do not Edit
 * @FilePath: \my-components\packages\input\index.js
 */
import CjrInput from "./src/input.vue"

CjrInput.install = (Vue) => {
	Vue.component(CjrInput.name, CjrInput)
}

export default CjrInput
