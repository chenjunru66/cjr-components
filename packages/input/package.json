{
  "name": "cjr-input",
  "version": "0.2.0",
  "description": "cjr-input component",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "del": "rimraf dist"
  },
  "keywords": [],
  "author": "chenjunru <963540300@qq.com>",
  "license": "MIT",
  "dependencies": {
    "lodash": "3"
  },
  "gitHead": "5ab89e77c41268b784ff72997171dbbd757e0eef"
}
