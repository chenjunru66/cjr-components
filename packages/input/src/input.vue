<!--
 * @Author: chenjunru
 * @Date: 2022-12-25 14:45:35
 * @LastEditTime: 2022-12-25 17:44:54
 * @LastEditors: chenjunru
 * @Description: Do not Edit
 * @FilePath: \my-components\packages\input\src\input.vue
-->
<template>
	<div>
		<input v-bind="$attrs" :type="type" :value="value" @input="handleInput" />
	</div>
</template>

<script>
	export default {
		name: "CjrInput",
		inheritAttrs: false,
		props: {
			value: {
				type: String,
			},
			type: {
				type: String,
				default: "text",
			},
		},
		methods: {
			handleInput(evt) {
				this.$emit("input", evt.target.value)
				const findParent = (parent) => {
					while (parent) {
						if (parent.$options.name === "CjrFormItem") {
							break
						} else {
							parent = parent.$parent
						}
					}
					return parent
				}
				const parent = findParent(this.$parent)
				if (parent) {
					parent.$emit("validate")
				}
			},
		},
	}
</script>

<style></style>
