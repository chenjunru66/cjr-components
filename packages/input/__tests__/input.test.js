/*
 * @Author: chenjunru
 * @Date: 2022-12-25 14:45:35
 * @LastEditTime: 2022-12-25 17:44:24
 * @LastEditors: chenjunru
 * @Description: Do not Edit
 * @FilePath: \my-components\packages\input\__tests__\input.test.js
 */
import input from "../src/input.vue"
import { mount } from "@vue/test-utils"

describe("cjr-input", () => {
	test("input-text", () => {
		const wrapper = mount(input)
		expect(wrapper.html()).toContain('input type="text"')
	})

	test("input-password", () => {
		const wrapper = mount(input, {
			propsData: {
				type: "password",
			},
		})
		expect(wrapper.html()).toContain('input type="password"')
	})

	test("input-password", () => {
		const wrapper = mount(input, {
			propsData: {
				type: "password",
				value: "admin",
			},
		})
		expect(wrapper.props("value")).toBe("admin")
	})

	test("input-snapshot", () => {
		const wrapper = mount(input, {
			propsData: {
				type: "text",
				value: "admin",
			},
		})
		expect(wrapper.vm.$el).toMatchSnapshot()
	})
})
