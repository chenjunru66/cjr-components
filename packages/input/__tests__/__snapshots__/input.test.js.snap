// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cjr-input input-snapshot 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
